nx/lib/nx.ex

```diff
@@ -4742,72 +4772,136 @@ defmodule Nx do
           ]
         ]
       >
+
+      iex> x = Nx.iota({2}, vectorized_axes: [x: 2])
+      iex> y = Nx.iota({3}, vectorized_axes: [y: 2])
+      iex> [broadcast_x, {broadcast_y}] = Nx.broadcast_vectors([x, {y}])
+      iex> broadcast_x
+      #Nx.Tensor<
+        vectorized[x: 2][y: 2]
+        s64[2]
+        [
+          [
+            [0, 1],
+            [0, 1]
+          ],
+          [
+            [0, 1],
+            [0, 1]
+          ]
+        ]
+      >
+      iex> broadcast_y
+      #Nx.Tensor<
+        vectorized[x: 2][y: 2]
+        s64[3]
+        [
+          [
+            [0, 1, 2],
+            [0, 1, 2]
+          ],
+          [
+            [0, 1, 2],
+            [0, 1, 2]
+          ]
+        ]
+      >
   """
   @doc type: :shape
   def broadcast_vectors([t]), do: [to_tensor(t)]
 
-  def broadcast_vectors(tensors) when is_list(tensors) do
-    {devectorized_tensors, target_vectorized_axes, offset} = do_reshape_vectors(tensors)
+  def broadcast_vectors(tensors_or_containers) when is_list(tensors_or_containers) do
```

Comment:
As always, we can split these changes into a separate PR. Just added them as needed here